File: iOS/PricetarPartner/PricetarPartner/Http/ApiRequester.swift
//
//  ApiRequester.swift
//  PricetarPartner
//
//  Created by Leapfrog-Inc. on 2021/02/21.
//

import Foundation

class ApiRequester {
    
    class func post(params: [String: String]?, completion: @escaping ((Bool, Any?) -> ())) {
        
        HttpRequester.post(url: Constants.ServerApiUrl, params: params) { result, data in
            if result, let data = data {
                do {
                    if let json = try JSONSerialization.jsonObject(with: data, options: JSONSerialization.ReadingOptions.allowFragments) as? Dictionary<String, Any> {
                        completion(true, json)
                        return
                    }
                } catch {}
            }
            completion(false, nil)
        }
    }
}

File: Android/app/src/main/java/jp/co/lfg/pricetarpartner/Http/ApiRequester.java
package jp.co.lfg.pricetarpartner.Http;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.HashMap;
import java.util.StringJoiner;

import jp.co.lfg.pricetarpartner.System.Constants;

public class ApiRequester {

    public static void post(HashMap<String, String> params, final Callback callback) {

        HttpRequester http = new HttpRequester(HttpRequester.RequestType.string, new HttpRequester.Callback() {
            @Override
            public void didReceiveData(boolean result, Object data) {
                if (result) {
                    try {
                        JSONObject json = new JSONObject((String)data);
                        callback.didReceiveData(true, json);
                        return;
                    } catch (JSONException e) { }
                }
                callback.didReceiveData(false, null);
            }
        });

        StringJoiner paramJoiner = new StringJoiner("&");
        for (String key : params.keySet()) {
            paramJoiner.add(key + "=" + params.get(key));
        }
        http.execute(Constants.ServerApiUrl, "POST", paramJoiner.toString());
    }

    public interface Callback {
        void didReceiveData(boolean result, JSONObject json);
    }
}
